homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,291 @@
+    ),
+    CurrentSensorEntityDescription(
+        key="current",
+        name="Current",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+    ),
+    InstantPowerSensorEntityDescription(
+        key="instant_power",
+        name="Instant Power",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=POWER_KILO_WATT,
+    ),
+    SessionEnergySensorEntityDescription(
+        key="session_energy",
+        name="Session Energy",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    TemperatureSensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+    ),
+    TotalChargedEnergySensorEntityDescription(
+        key="total_charged_energy",
+        name="Total Charged Energy",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    VoltageSensorEntityDescription(
+        key="voltage",
+        name="Voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+    ),
+    InstallCurrentSensorEntityDescription(
+        key="install_current",
+        name="Install Current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+    ),
+    DynamicCurrentSensorEntityDescription(
+        key="dynamic_current",
+        name="Dynamic Current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+    ),
+    LedMaxBrightnessSensorEntityDescription(
+        key="led_max_brightness",
+        name="Led Brightness",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+
+    sensors = [
+        LektricoSensor(
+            sensor_desc,
+            _lektrico_device,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for sensor_desc in SENSORS
+    ]
+
+    async_add_entities(sensors, False)
+
+
+class LektricoSensor(CoordinatorEntity, SensorEntity):
+    """The entity class for Lektrico charging stations sensors."""
+
+    entity_description: LektricoSensorEntityDescription
```

Comment:
```suggestion
    entity_description: LektricoSensorEntityDescription
    _attr_has_entity_name = True
```

Reply:
If I add this  _attr_has_entity_name = True  , in Entities,  one sensor's name is correct, the all others have the friendly name duplicated. 
ex: friendly name = asd
sensor name = asd asd Current
sensor entity id = sensor.asd_asd_current

Reply:
I made the changes to use modern entity naming, thanks.